vehicle/mkvi/software/bms/bms.yml

```diff
@@ -0,0 +1,299 @@
+          type: uint16_t
+          name: V
+          offset: 0
+          scale: 5 / 1024
+
+  - name: bms_metrics
+    id: 0x510
+    freq_hz: 100
+    signals:
+      - name: voltage_pec_error_count
+        slice: 0 + 8
+        unit:
+          type: uint8_t
+          name: "errors"
+          scale: 1
+          offset: 0
+      - name: temperature_pec_error_count
+        slice: 8 + 8
+        unit:
+          type: uint8_t
+          name: "errors"
+          scale: 1
+          offset: 0
+
+  - name: bms_debug
+    id: 0x511
+    freq_hz: 100
+    signals:
+      - name: dbg_1
+        slice: 0 + 16
+        unit:
+          type: uint16_t
+          name: ""
+          scale: 1
+          offset: 0
+      - name: dbg_2
+        slice: 16 + 16
+        unit:
+          type: uint16_t
+          name: ""
+          scale: 1
+          offset: 0
+      - name: dbg_3
+        slice: 32 + 16
+        unit:
+          type: uint16_t
+          name: ""
+          scale: 1
+          offset: 0
+      - name: dbg_4
+        slice: 48 + 8
+        unit:
+          type: uint16_t
+          name: ""
+          scale: 1
+          offset: 0
+      - name: charger_hardware_fault
+        slice: 56 + 1
+        unit:
+          type: bool
+          values:
+            t: FAULT
+            f: Normal
+      - name: charger_temperature_protection
+        slice: 57 + 1
+        unit:
+          type: bool
+          values:
+            t: FAULT
+            f: Normal
+      - name: charger_input_voltage
+        slice: 58 + 1
+        unit:
+          type: bool
+          values:
+            t: FAULT
+            f: Normal
+      - name: charger_starting_state
+        slice: 59 + 1
+        unit:
+          type: bool
+          values:
+            t: OFF
+            f: On
+      - name: charger_communication
+        slice: 60 + 1
+        unit:
+          type: bool
+          values:
+            t: FAULT
+            f: Normal
+
+  - name: bms_voltage
+    id: 0x401
+    freq_hz: 4
+    signals:
+      - name: ic
+        slice: 0 + 8
+        unit:
+          type: uint8_t
+          name: ""
+          scale: 1
+          offset: 0
+      - name: cell
+        slice: 8 + 8
+        unit:
+          type: enum
+          values:
+            - REG_A
+            - REG_B
+            - REG_C
+            - REG_D
+      - name: voltage_1
+        slice: 16 + 16
+        unit:
+          type: uint16_t
+          name: "volts"
+          scale: 0.0001
+          offset: 0
+      - name: voltage_2
+        slice: 32 + 16
+        unit:
+          type: uint16_t
+          name: "volts"
+          scale: 0.0001
+          offset: 0
+      - name: voltage_3
+        slice: 48 + 16
+        unit:
+          type: uint16_t
+          name: "volts"
+          scale: 0.0001
+          offset: 0
+
+  - name: bms_temperature
+    id: 0x402
+    freq_hz: 4
+    signals:
+      - name: ics
+        slice: 0 + 8
+        unit:
+          type: enum
```

Comment:
I think this message will need to change structure with the new temperature architecture. Let's chat about this in person.